sopel/test_tools.py

```diff
@@ -259,3 +272,257 @@ def run_example_tests(filename, tb='native', multithread=False, verbose=False):
         args.extend(["-n", str(cpu_count())])
 
     pytest.main(args)
+
+
+def botfactory(settings, autoloads=None):
+    """Build a test instance of Sopel, with a :class:`MockIRCBackend`.
+
+    :param settings: Sopel's configuration for testing purposes
+    :type settings: :class:`sopel.config.Config`
+    :param list autoloads: list of plugins to autoload after creating the bot
+    :return: a test instance of the bot
+    :rtype: :class:`sopel.bot.Sopel`
+
+    This will instanciate a :class:`~sopel.bot.Sopel` object, replace its
+    backend with a :class:`~MockIRCBackend`, and then autoload plugins. This
+    will automatically load the ``coretasks`` plugins, and every other plugins
```

Comment:
Two-for-one special on nemese**S**!

```suggestion
    will automatically load the ``coretasks`` plugin, and every other plugin
```